blockindex/liquidstaking_cache.go

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) 2023 IoTeX Foundation
+// This source code is provided 'as is' and no warranties are given as to title or non-infringement, merchantability
+// or fitness for purpose and, to the extent permitted by law, all liability for your use of the code is disclaimed.
+// This source code is governed by Apache License 2.0 that can be found in the LICENSE file.
+
+package blockindex
+
+import (
+	"math/big"
+	"time"
+
+	"github.com/iotexproject/iotex-core/db/batch"
+)
+
+type (
+	liquidStakingCache struct {
+		idBucketMap           map[uint64]*BucketInfo     // map[token]BucketInfo
+		candidateBucketMap    map[string]map[uint64]bool // map[candidate]bucket
+		idBucketTypeMap       map[uint64]*BucketType     // map[token]BucketType
+		propertyBucketTypeMap map[int64]map[int64]uint64 // map[amount][duration]index
+		height                uint64
+	}
+)
+
+func newLiquidStakingCache() *liquidStakingCache {
+	return &liquidStakingCache{
+		idBucketMap:           make(map[uint64]*BucketInfo),
+		idBucketTypeMap:       make(map[uint64]*BucketType),
+		propertyBucketTypeMap: make(map[int64]map[int64]uint64),
+		candidateBucketMap:    make(map[string]map[uint64]bool),
+	}
+}
+
+func (s *liquidStakingCache) writeBatch(b batch.KVStoreBatch) error {
```

Comment:
we don't need this func
all the delta/changes are already captured in the cache itself, no need to `deserialize()` again from the items in the batch
see comment where it is being called

Reply:
use `merge` from delta instead